run_tests.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python3
+
+'''
+
```

Comment:
Can you please include some detail here about what this script does? When is it run? What triggers it? 

Just a basic "the purpose of this script is to ...."

And a basic overview of what it does - the steps it takes.

Anything to make it easier to maintain when no-one has touched it for 6 months and then has to come back to tweak something.